knimin/lib/data_access.py

```diff
@@ -2469,88 +2482,83 @@ def _study_is_unique(self, qiita_study_id=None, title=None, skip_id=None):
+            if res:
+                raise ValueError("Study (%s, %s) conflicts with studies %s"
+                                 % (qiita_study_id, title,
+                                    ', '.join(map(str, res))))
+
+    def create_study(self, qiita_study_id, title, alias, jira_id):
         """Creates a study
 
         Parameters
         ----------
-        qiita_study_id : int, optional
-            Assigns a Qiita study ID to the study
-        title : str, optional
-            Assigns a title to the study
-            Either qiita_study_id or title must be specified
-        alias : str, optional
-            Assigns an alias to the study
-        notes : str, optional
-            Makes notes of the study
-
-        Returns
-        -------
-        int
-            ID of the study created
+        qiita_study_id : int
+            The Qiita study ID
+        title : str
+            The title of the study
+        alias : str
+            The alias of the study
+        jira_id : str
+            The study JIRA key
         """
         with TRN:
             self._study_is_unique(qiita_study_id, title)
-            sql = """INSERT INTO pm.study (qiita_study_id, title, alias, notes)
-                     VALUES (%s, %s, %s, %s)
-                     RETURNING study_id"""
-            sql_args = [qiita_study_id, title, alias, notes]
+            sql = """INSERT INTO pm.study (study_id, title, alias, jira_id)
+                     VALUES (%s, %s, %s, %s)"""
+            sql_args = [qiita_study_id, title, alias, jira_id]
             TRN.add(sql, sql_args)
-            return TRN.execute_fetchlast()
+            TRN.execute()
 
-    def edit_study(self, study_id, qiita_study_id=None, title=None, alias=None,
-                   notes=None):
+    def edit_study(self, study_id, title=None, alias=None):
         """Edits properties of an existing study
 
         Parameters
         ----------
         study_id : int
             ID of the study to edit
-        qiita_study_id : int, optional
-            Assigns a Qiita study ID to the study
         title : str, optional
             Assigns a title to the study
             Either qiita_study_id or title must be specified
         alias : str, optional
             Assigns an alias to the study
-        notes : str, optional
-            Makes notes of the study
+
+        Raises
+        ------
+        ValueError
+            If both title and alias are None
         """
+        if title is None and alias is None:
+            raise ValueError(
+                "At least one of title or alias should be provided")
+
         with TRN:
             self._study_exists(study_id)
-            self._study_is_unique(qiita_study_id, title, study_id)
+            self._study_is_unique(study_id, title, study_id)
+            cols = []
+            sql_args = []
+            if title is not None:
+                cols.append("title = %s")
+                sql_args.append(title)
```

Comment:
These lines are creating the list of columns to be modified and the parameters passed. Note that `cols` is used in line 2561 in the `join` call, and `sql_args` is used when adding the call to the transaction object in line 2562.